src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -6250,21 +6250,24 @@ private sealed class NamespaceCompletion : TypeCompletionBase
         {
             internal string Namespace;
 
-            internal override CompletionResult GetCompletionResult(string keyMatched, string prefix, string suffix)
+            internal override CompletionResult GetCompletionResult(string keyMatched, string prefix, string suffix, string namespaceToRemove)
             {
+                string completion = string.IsNullOrEmpty(namespaceToRemove)
+                    ? $"{prefix}{Namespace}{suffix}"
+                    : $"{prefix}{Namespace.Substring(namespaceToRemove.Length + 1)}{suffix}";
```

Comment:
I think it is more correct to use InvariantCulture.

Reply:
I don't understand. How do you want me to use InvariantCulture?

Reply:
.Net 6.0 has new overload  `public static string Create(IFormatProvider? provider, [InterpolatedStringHandlerArgument("provider")] ref DefaultInterpolatedStringHandler handler)`
and now we can do:
```c#
                string completion = string.IsNullOrEmpty(namespaceToRemove)
                    ? string.Create(CultureInfo.InvariantCulture, $"{prefix}{Namespace}{suffix}")
                    : string.Create(CultureInfo.InvariantCulture, $"{prefix}{Namespace.Substring(namespaceToRemove.Length + 1)}{suffix}");
```

Reply:
What's the advantage of that? Better performance? I remember reading something about the `$"Hello {var}"` syntax getting optimized in .NET 6 so is this even faster?

Reply:
If we say about the PR and PowerShell all code must use InvariantCulture (excluding format output).

If we say about .Net, yes, .Net 6.0 got  optimizations for interpolated strings.

Reply:
@MartinGC94 I think the guiding principle here is that core language behavior should be 100% deterministic regardless of platform locale.

Reply:
If I search the entire solution for `string.Create` I find 4 instances, one of them is just a comment and the 3 other instances don't do anything with the Culture.  
If I search for `$"` I find 147 matches, some of them are false positives from regex patterns but for the most part they are just normal strings with variables that look like my code.

I'm not against changing it, I'm just trying to understand the logic. Should all strings in this project be created like that in the future? If not, how do I know when to do it and when not to do it? What's the benefit? Clearly it's not strictly necessary since PowerShell works fine across multiple locales today.

Reply:
> If I search the entire solution for `string.Create` I find 4 instances, one of them is just a comment and the 3 other instances don't do anything with the Culture.

I say about new overload - we did not use it before.

>If I search for $" I find 147 matches, some of them are false positives from regex patterns but for the most part they are just normal strings with variables that look like my code.

I hope all they for formatting output to users. 

Logic is simple - PowerShell language and scripts must be deterministic regardless of platform locale as @IISResetMe said, but data output should be in user locale as users expect.

Reply:
The argument doesn't make any sense to me. You are telling me 3 contradicting things:

1. PowerShell should be 100% deterministic, regardless of platform locale. The implication here is that PowerShell is already 100% deterministic today.
2. This new overload is necessary to make strings in PowerShell 100% deterministic, regardless of platform locale.
3. This overload isn't used anywhere else because it's new.

These 3 can't all be true at the same time. Either PowerShell is "broken" today across different locales or it's working fine and this new overload isn't actually necessary. Which one is it?

Reply:
@MartinGC94 Remember we're talking about "what the ideal world looks like" vs "what the world actually looks like" here. The fact that PowerShell _should_ satisfy some arbitrary constraint (like "locale should not impact runtime behavior") does not mean the current implementation _does_ so under all circumstances. 

@iSazonov's suggestion would bring us _closer_ to guaranteeing the ideal, that's all.

Reply:
Ok, I've made the change. I still don't understand why strings need culture specific info unless you are somehow converting the string (upper/lower, numbers, dates, etc.) but I can look that up myself at some point.

Reply:
Using `string.Format()` overload with _explicit_ culture always was our best practice - with InvariantCulture in Engine and CurrentCulture in formatting for user output.
By default interpolated strings use CurrentCulture. Now we have new more efficient API for interpolated strings where we can use explicit Culture too.